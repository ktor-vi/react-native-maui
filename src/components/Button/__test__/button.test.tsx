import * as React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native';
import { Button } from '../index';

describe('Test:Button', () => {
  it('render correctly', () => {
    const tree = render(
      <Button onPress={() => {}}>
        <Text>默认样式</Text>
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('base button', async () => {
    const onPressMock = jest.fn();
    const { queryByText: queryByText1 } = render(
      <Button onPress={onPressMock}>
        <Text>默认样式</Text>
      </Button>
    );
    const element1 = queryByText1('默认样式');
    expect(element1).not.toBeNull();

    // onPressMock被调用了
    fireEvent(element1, 'onPress'); // 触发Button里的组件方法
    expect(onPressMock).toHaveBeenCalled();
  });

  it('button disabled', async () => {
    const onPressMock2 = jest.fn();
    const { queryByText: queryByText2 } = render(
      <Button onPress={onPressMock2} disabled>
        <Text>禁止点击</Text>
      </Button>
    );

    const element2 = queryByText2('禁止点击');
    expect(element2).not.toBeNull();

    // 不可点击
    fireEvent(element2, 'onPress');
    expect(onPressMock2).not.toHaveBeenCalled();
  });
});
