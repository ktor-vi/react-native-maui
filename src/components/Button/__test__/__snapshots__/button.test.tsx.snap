// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test:Button render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#D9D9D9",
      "borderRadius": 3,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Text>
    默认样式
  </Text>
</View>
`;
